### PR TITLE
[Fix-3457][flink] fix flink args build problem

### LINKED ISSUE #3987: [Bug]use dolphinscheduler1.3.1 scheduler flink1.11 - Could not build the program from JAR file: JAR file does not exist: -yn
dolphinscheduler1.3.1

flink1.11

use dolphinscheduler1.3.1 scheduler flink1.11 
the log error is : -yn 
Could not build the program from JAR file: JAR file does not exist: -yn

Use the help option (-h or --help) to get help on the command.

the flink1.10 remove the -yn and -yST parameters

Which dolphinscheduler version  will support the flink1.11??

### PR DESCRIPTION
**Pull request by chengshiwen**
## *Tips*
- *Thanks very much for contributing to Apache DolphinScheduler.*
- *Please review https://dolphinscheduler.apache.org/en-us/community/index.html before opening a pull request.*

## What is the purpose of the pull request

*Fix flink args build problem* #3457 

this closes #3987 #3457 

The official flink run command is `flink run [OPTIONS] <jar-file> <arguments>` (ref: [flink-1.9-cli](https://ci.apache.org/projects/flink/flink-docs-release-1.9/ops/cli.html) and  [flink-1.11-cli](https://ci.apache.org/projects/flink/flink-docs-release-1.11/ops/cli.html)), but there are some errors in the code:
 - the `other args` is after the `main args`, so that it will be part of the `<arguments>`
 - the flink queue option is also after the `main args`, so that the option doesn't work
 - the flink queue option is `-yqu`, not `--qu`, and it's only for yarn. see [flink-1.9-cli](https://ci.apache.org/projects/flink/flink-docs-release-1.9/ops/cli.html) and  [flink-1.11-cli](https://ci.apache.org/projects/flink/flink-docs-release-1.11/ops/cli.html)
 - the comment `--files --conf --libjar ...` is for spark, not for flink

## Brief change log

*Flink args:*
 - *[Fix][Flink] fix flink args build problem*
   - Adjust the order of the `other args` and optimize the logic
   - Rename the flink queue option `--qu` to `-yqu`
 - *[Fix][Flink] fix FlinkArgsUtilsTest*
 - *[Improvement][UI] hide version and cluster input when deployMode is local*

## Verify this pull request

*(Please pick either of the following options)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(and)

This change added tests and can be verified as follows:

Assume that:
 - main args: `--input file:///home/user/hamlet.txt --output file:///home/user/wordcount_out`
- other args: `-p 4`

**Before fix, the generated flink commands are as follow**:
 - **local**: `flink run -c org.apache.flink.examples.java.wordcount.WordCount WordCount.jar --input file:///home/user/hamlet.txt --output file:///home/user/wordcount_out -p 4`
 - **cluster**: `flink run -m yarn-cluster -ys 1 -yn 2 -yjm 1G -ytm 2G -d -c org.apache.flink.examples.java.wordcount.WordCount WordCount.jar --input file:///home/user/hamlet.txt --output file:///home/user/wordcount_out --qu test -p 4`

**After fix, the generated flink commands are as follow**:
 - **local**: `flink run -p 4 -c org.apache.flink.examples.java.wordcount.WordCount WordCount.jar --input file:///home/user/hamlet.txt --output file:///home/user/wordcount_out`
 - **cluster**: `flink run -m yarn-cluster -ys 1 -yn 2 -yjm 1G -ytm 2G -yqu test -d -p 4 -c org.apache.flink.examples.java.wordcount.WordCount WordCount.jar --input file:///home/user/hamlet.txt --output file:///home/user/wordcount_out`

For the commit *[Improvement][UI] hide version and cluster input when deployMode is local*:
image

### CODE REVIEW

File: dolphinscheduler-api/src/main/resources/application-api.properties [hunk 1]
```diff
@@ -35,7 +35,7 @@ server.compression.enabled=true
 server.compression.mime-types=text/html,text/xml,text/plain,text/css,text/javascript,application/javascript,application/json,application/xml
 
 #post content
-server.jetty.max-http-post-size=5000000
```
Comment (chengshiwen) [inline dolphinscheduler-api/src/main/resources/application-api.properties:38]: `server.jetty.max-http-post-size` is just deprecated

**Comment by lgcareer** [comment]
@chengshiwen Hi,Could you please list the issue fixed by this pr?

**Comment by chengshiwen** [comment]
@Igcareer This bug appeared in our business, so I don't check the issues.
Now I find that the Issue number related to this Pull Request is [#3457](https://github.com/apache/incubator-dolphinscheduler/issues/3457)

**Review by CalvinKirs — APPROVED** [review]
LGTM(server code)

**Review by dailidong — APPROVED** [review]
+1
great job, [666]

**Comment by dailidong** [comment]
@xingchun-chen , I put this PR to milestone 1.3.4-release,  could you do this bug fix test?

**Comment by chuyang-zl** [comment]
Please tell me if this change has been incorporated into version 1.3.4. I did not see this functionality incorporated in the code for the 1.3.4-release branch

---

请问这个改动是否合入了1.3.4版本，我在1.3.4-release分支的代码中没有看到此功能的合入

**Comment by chengshiwen** [comment]
@chuyang-zl Not yet, maybe the commit can be cherry-picked into the next release 1.3.5?
@lgcareer @dailidong

**Comment by CalvinKirs** [comment]
Thank you for your feedback, we will release it in the next version.

**Pull request merged**